### ISSUE TITLE
configure feature using file

### ISSUE THREAD

**Issue by dmail** [dmail-old/jsenv-core#20]
write stuff like import string/prototype/at instead of having an array of feature or query params?

**Comment by dmail** [comment]
The concept would be to use systemjs builder (which use rollup) being able to remove some import resulting in the inclusion of polyfill.js inside the systemjs build.

**Comment by dmail** [comment]
Migrating to systemjs builder is a BIG change and is challenging in many ways.
I'll stick to rollup for generating the polyfill.js but I'll still do the following changes to improve how it works:

- The root folder when using rollup will be `/` and not `/src/features` anymore
- `fix-helpers.js` will be moved to sthing like `util/fix/helpers.js` (same for test-helpers.js)
- all jsenv.js code will be splitted into files written using same constraints than features files
- to specify the features you need you'll write `import '@jsenv/features/object-assign.js'` instead of passing an array of required features.

**Comment by dmail** [comment]
I **have to** use systemjs-builder, at least I have to produce `System.register` output so that code not depending on polyfill.js build can reuse code bundled inside it.

First step is to list what features I do need (fetching dependencies, conditional import etc) with systemjs and later find how to do it.

**Comment by dmail** [comment]
This
```javascript
builder.config({
  transpiler: false,
  'meta': {
    '*': {format: 'system'}
  }
});
```
Combined with a custom fetch that would transpile to system format + do babel + optional minification we would have an output which use our transpiler and our cache mecanism.

Concerning the conditional import (some import must be discarded depending on the test/fix status). It means we would have to transform the source, parse it, readdependencies and for the import statements which are discarded, remove them from source.
Nope, in fact thoose statements MUST be discarded from source (or at least be ignored by the builder using conditional import) because build abstract source is in charge to import them when needed.

Basically writing sthing like `import '@jsenv-features/object/assign/test.js?#CONST|ignore' so that we know it depends on object-assign but it's not imported by systemjs builder.

**Comment by dmail** [comment]
I'm tired by this issue... I can't make systemjs-builder behave as I need.

I can try to "rewrite" systemjs builder, basically it means being able to.

- Fetch dependencies (currently my way of doing this is dumb because it doesn't cache like systems builder does (does it?))
- Concatenate source & sourcemap of all modules (may cause issue but should be ok)
- Generate System.register calls (calls must be named)

**Comment by dmail** [comment]
ORRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRRR just using systemjs builder and doing 
`export const dependencies = [];` ?

or keeping import of dependencies but removing them with a babel plugin ?

**Comment by dmail** [comment]
Lol let's just close this just lol.
